src/main.ts

```diff
@@ -450,6 +451,42 @@ export default class ImageWindow extends Plugin {
             })
         );
 
+        this.registerDomEvent(document, "contextmenu", (event: MouseEvent) => {
+            const target = event.target as HTMLElement;
+            if (target.localName !== "img") return;
+
+            const vaultName = this.app.vault.getName().replace(" ", "%20")
+            const imgPath = (target as HTMLImageElement).currentSrc;
+            const cleanPath = imgPath.split(`${vaultName}/`)[1].split("?")[0];
+            const file = this.app.vault.getAbstractFileByPath(cleanPath);
```

Comment:
There is a private method on the Vault to get the TFile from a resource path:

`vault.resolveFileUrl(path)`